A6.md

```diff
@@ -0,0 +1,398 @@
+
+The only RPCs that are counted as failures for the throttling policy are RPCs that fail with a status code that qualifies as a retryable or non-fatal status code (see [here](#retryable-status-codes) and [here](#hedging-policy)), or that receive a pushback response indicating not to retry. This avoids conflating server failure with responses to malformed requests (such as the `INVALID_ARGUMENT` status code).
+
+#### Pushback
+Servers may explicitly pushback by setting metadata in their response to the client. The pushback can either tell the client to retry after a given delay or to not retry at all. If the client has already exhausted its `max_retry_attempts`, the call will not be retried even if the server says to retry after a given delay.
+
+A new metadata key, `"x-grpc-retry-pushback-ms"`, will be added to support server pushback. If the value for pushback is set to -1, then it will be seen as the server asking the client not to retry at all.
+
+#### Summary of Retry and Hedging Logic
+There are five possible types of server responses. The list below enumerates the behavior of retry and hedging policies for each type of response. In all cases, if the maximum number of retry attempts or the maximum number of hedged requests is reached, no further RPCs are sent. Hedged RPCs are returned to the client application when all outstanding and pending requests have either received a response or been canceled.
+
+1. OK
+  1. Retry policy: Successful response, return success to client application
+  2. Hedging policy: Successful response, cancel previous and pending hedges
+2. Retryable/Non-Fatal Status Code
+  1. Retry policy: Retry according to policy
+  2. Hedging policy: Immediately send next scheduled hedged request, if any. Subsequent hedged requests will resume at `hedging_delay`
+3. Fatal Status Code
+  1. Retry policy: Don't retry, return failure to client application
+  2. Hedging policy: Cancel previous and pending hedges
+4. Pushback: Don't retry
+  1. Retry policy: Don't retry, return failure to client application
+  2. Hedging policy: Don’t send any more hedged requests.
+5. Pushback: Retry in *n* ms
+  1. Retry policy: Retry in *n* ms. If this attempt also fails, retry delay will reset to initial backoff for the following retry (if applicable)
+  2. Hedging policy: Send next hedged request in *n* ms. Subsequent hedged requests will resume at `n + hedging_delay`
+
+![State Diagram](A6_graphics/StateDiagram.png)
+
+### Retry Internals
+
+#### Where Retries Occur
+
+The retry policy will be implemented in-between the channel and the load balancing policy. That way every retry gets a chance to be sent out on a different subchannel than it originally failed on.
```

Comment:
This is a good point, but currently the metadata only says if the RPC is a retry. We are considering adding information about the previously used subchannel(s), based on comments from the gRFC email thread. If/once that is added, then adding the link you suggest (based on the retry metadata...) makes sense.